providers/ldap: use RDN when using posixGroup's memberUid attribute

asteven opened the pull request:
# Details
* **Does this resolve an issue?**
Resolves #1436

## Changes
### New Features
* Uses the RDN instead of the FDN when establishing group memberships based on posixGroup's 'memberUid' attribute.

BeryJu commented:
LGTM, I think all it needs is a black pass, and the rest of the CI tests should be fixed when you rebase on master

asteven committed:
providers/ldap: use RDN when using posixGroup's memberUid attribute

Use the RDN instead of the FDN when establishing group memberships based on posixGroup's 'memberUid' attribute.

fixes #1436

Signed-off-by: Steven Armstrong <steven@armstrong.cc>

asteven commented:
> LGTM, I think all it needs is a black pass, and the rest of the CI tests should be fixed when you rebase on master

Black now passing. Rebased and squashed.
